src/dataDisplay/table.jsx

```diff
@@ -1,21 +1,522 @@
-'use strict';
-
-import React, { Component } from 'react';
 import ClassNames from 'classnames';
+import {
+    debounce,
+    get,
+} from 'lodash';
 import PropTypes from 'prop-types';
+import React from 'react';
+import ScrollBar from 'react-custom-scrollbars';
+import { withStyles } from '../styles';
+import Utils from '../utils/utils';
 import TableBody from './tableBody';
 import TableCell from './tableCell';
+import {
+    DEBOUNCE_WAIT_TIME,
+    TH_RESIZABLE_MAX_WIDTH,
+} from './tableConstants';
 import TableHeader from './tableHeader';
-import TableHeaderCell from './tableHeaderCell';
 import TableRow from './tableRow';
+import TableHeaderCell from './tableHeaderCell';
+
+const propTypes = {
+    /**
+     * Basic (default) styling.
+     */
+    basic: PropTypes.bool,
+    /**
+     * If `true`, all columns will be divided.
+     */
+    celled: PropTypes.bool,
+    /**
+     * The content of the Table.
+     */
+    children: PropTypes.node,
+    /**
+     * Override or extend the styles applied to Table.
+     */
+    classes: PropTypes.shape({
```

Comment:
Hmm, this description feels slightly misleading to me.  This is actually _built-in_ classes/styling that is _not_ really end-user customizable, because these are the classes added by the `withStyles()` HOC, right?

@morethanfire Could you perhaps suggest a better, standard description for the `classes` prop for this and similar components?

If push comes to shove, this is perhaps a NAB in grand scheme of things.

Reply:
The description is correct. We can pass in another class object that overrides the other class. It's a little iterative with this one though, because there isn't a `root` and etc. 

.e.g.
```
<Table
    classes={{
        root: classes.root,
    }}
>
    ...
</Table>
```

Reply:
Huh, didn't realize that.  So `withStyles()` will combine an existing `classes` prop object with the one generated by `useStyles()` function?  Interesting.

In any event, resolving this thread.